### PR TITLE
CLOUDP-271909: fix deleted user struct

### LINKED ISSUE #239: UsersDeleted field of Automation Config does not work with MongoDBUser struct
Hi, 

In order to remove a mongodb user, I want to remove the user from the UsersWanted list and add it to the UsersDeleted list of the AutomationConfig.Auth. 

The UsersDeleted list requires the MongoDBUser struct. However, if I try to add a user object containing only the username and the database to this list, I receive this error: invalid attribute authenticationRestrictions specified (although the field is set to nil). 

After reading this documentation https://www.mongodb.com/docs/ops-manager/current/reference/api/automation-config/automation-config-parameters/ I would suggest a specific MongoDBUserDeleted struct with the fields described in the documentation for the UsersDeleted list. 

I forked the repository and made a solution proposal there: https://github.com/mongodb/go-client-mongodb-ops-manager/compare/master...Seli19972:go-client-mongodb-ops-manager:master

Let me know what you think! 

Thanks in advance!

Best regards,
Selina

### PR DESCRIPTION
**Pull request by husniMDB**
## Proposed changes

_Jira ticket:_  CLOUDP-271909

Closes #239 

## Checklist

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by bdaw — DISMISSED** [review]
lgtm

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: lint
         uses: golangci/golangci-lint-action@v6.1.0
         with:
-          version: v1.54.2
+          version: v1.60
```
Comment (gssbzn) [inline .github/workflows/pr.yml:22]: also update https://github.com/mongodb/go-client-mongodb-ops-manager/blob/221d95ead25f991999b71da8d21ab66a2b0591ec/Makefile#L4

File: opsmngr/automation_config.go [hunk 2]
```diff
@@ -231,6 +231,12 @@ type MongoDBUser struct {
 	Username                   string                      `json:"user"` //nolint:tagliatelle // Username is a better name than just user
 }
 
+// MongoDBDeletedUser user to delete from DBs list.
+type MongoDBDeletedUser struct {
+	Username  string   `json:"user"` //nolint:tagliatelle // used as in the API
+	Databases []string `json:"dbs"`  //nolint:tagliatelle // used as in the API
```
Comment (gssbzn) [inline opsmngr/automation_config.go:237]: let's follow tagliatelle
```suggestion
	User  string   `json:"user"` //nolint:tagliatelle // used as in the API
	DBs []string `json:"dbs"`  //nolint:tagliatelle // used as in the API
```

Comment (husniMDB) [reply opsmngr/automation_config.go]: had to keep //nolint for dbs as it's not accepted this way and needs to be dBs

**Review by gssbzn — APPROVED** [review]

**Pull request merged**